visidata/loaders/_pandas.py

```diff
@@ -49,6 +69,57 @@ def sort(self):
             ascending += [not reverse] * len(cols)
         self.rows.sort_values(by=by_cols, ascending=ascending, inplace=True)
 
+    def _checkSelectedIndex(self):
+        if self._selectedMask.index is not self.df.index:
+            # self.df was modified inplace, so the selection
+            # is no longer valid -- just delete it. Maybe too heavy but erring
+            # on side of correctness here.
+            vd.status('pd.DataFrame.index updated, clearing selected {} rows'
+                      .format(self._selectedMask.sum()))
```

Comment:
Let's use `.nSelected` as you've redefined below. and a minor text change, to `{} selected rows`.

Reply:
It would lead to a cyclic reference here, since `nSelected` is calling `_checkSelectedIndex` itself, although the calling of `_checkSelectedIndex` is arguably overly defensive.